### PR TITLE
Chore: Centralize email validation functionality

### PR DESCRIPTION
**Pull request by KevLehman**
## Proposed changes (including videos or screenshots)

- Create lib for validating emails
- Modify places that validate emails to use the new central function

### CODE REVIEW

File: lib/emailValidator.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	const basicEmailRegex = /^.+@.+$/;
+	const rfcEmailRegex = /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+
+	switch (options.style) {
+		case 'basic':
+			return basicEmailRegex.test(email);
+		case 'rfc':
+			return rfcEmailRegex.test(email);
+		default:
+			return basicEmailRegex.test(email);
```
Comment (tassoevan) [inline lib/emailValidator.ts:11]:
```suggestion
		case 'rfc':
			return rfcEmailRegex.test(email);
		case 'basic':
		default:
			return basicEmailRegex.test(email);
```

File: lib/emailValidator.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export const validateEmail = (email: string, options: { style: string } = { style: 'basic' }): boolean => {
+	const basicEmailRegex = /^.+@.+$/;
```
Comment (sampaiodiego) [inline lib/emailValidator.ts:2]: looking into some regex you modified to use this function I found this one interesting: `[^@].*@[^@]`, it makes sure only a single `@` is allowed.. wdyt? 👀

File: lib/emailValidator.ts (hunk 1 above)
Comment (KevLehman) [reply lib/emailValidator.ts:11]: I thought we had the no-fallthrough rule enabled 🤔

File: lib/emailValidator.ts (hunk 2 above)
Comment (KevLehman) [reply lib/emailValidator.ts:2]: image
Not sure 🙈 if we should use it tho

Comment (sampaiodiego) [reply lib/emailValidator.ts:2]: oopsie.. looks like I missread the regex 🙈

Comment (sampaiodiego) [reply lib/emailValidator.ts:2]: ok, this one 😬  `^[^@]+@[^@]+$`

**Review by sampaiodiego — APPROVED** [review]

**Comment by sampaiodiego** [comment]
Although I have approved it already, what about writing some unit tests for it? maybe we can have an agreement that all functions in `/lib` should have unit tests.. wdyt?

**Pull request merged**